compaction.go

```diff
@@ -1,16 +1,55 @@
 package lsm
 
 func (db *DB) compactLevel(level int) error {
-	if level == 0 {
+	if level == 0 && db.needL0Compaction() {
 		return db.compactLevel0()
+	} else if level > 0 && db.needLevelNCompaction(level) {
+		return db.compactLevelN(level)
 	}
-	return db.compactLevelN(level)
+	return nil
 }
 
 func (db *DB) compactLevel0() error {
+	iterators0 := make([]*SSTableIterator, 0)
+	for _, sstable := range db.levels[0].sstables {
+		iterators0 = append(iterators0, sstable.Iterator())
+	}
+	iterators1 := make([]*SSTableIterator, 0)
+	for _, sstable := range db.levels[1].sstables {
+		iterators1 = append(iterators1, sstable.Iterator())
+	}
+
+	iterators := append(iterators0, iterators1...)
+	sstList, err := db.mergeIterators(iterators, 1)
```

Comment:
### Runtime Errors
Handle potential runtime errors when merging iterators. Specifically, ensure that `db.mergeIterators(iterators, 1)` and `db.mergeIterators(iterators, level+1)` are checked for errors and handle them appropriately. Consider logging the error and returning a more informative error message:
```go
if err != nil {
    log.Printf("Error merging iterators: %v", err)
    return fmt.Errorf("failed to merge iterators: %w", err)
}
```